packages/base/src/delegate/ResizeHandler.js

```diff
@@ -1,65 +1,63 @@
-import CustomResize from "./CustomResize.js";
+import { nativeObserve, nativeUnobserve } from "./NativeResize.js";
+
+let observe = nativeObserve;
+let unobserve = nativeUnobserve;
+
+/**
+ * Set a callback to be executed whenever a DOM node needs to be observed for size change.
+ * @public
+ * @param callback
+ */
+const setResizeHandlerObserveCallback = callback => {
+	observe = callback;
+};
+
+/**
+ * Set a callback to be executed whenever a DOM node needs to no longer be observed for size changes
+ * @public
+ * @param callback
+ */
+const setResizeHandlerUnobserveCallback = callback => {
+	unobserve = callback;
+};
 
 class ResizeHandler {
-	static initialize() {
-		ResizeHandler.Implementation = window.ResizeObserver ? NativeResize : CustomResize;
-		ResizeHandler.Implementation.initialize();
-	}
-
-	/**
-	 * @static
-	 * @private
-	 * @param {*} ref Reference to be observed
-	 * @param {*} callback Callback to be executed
-	 * @memberof ResizeHandler
-	 */
-	static attachListener(ref, callback) {
-		ResizeHandler.Implementation.attachListener.call(ResizeHandler.Implementation, ref, callback);
-	}
-
-	/**
-	 * @static
-	 * @private
-	 * @param {*} ref Reference to be unobserved
-	 * @memberof ResizeHandler
-	 */
-	static detachListener(ref, callback) {
-		ResizeHandler.Implementation.detachListener.call(ResizeHandler.Implementation, ref, callback);
-	}
-
-
 	/**
 	 * @static
 	 * @public
-	 * @param {*} ref Reference to a UI5 Web Component or DOM Element to be observed
+	 * @param {*} element UI5 Web Component or DOM Element to be observed
 	 * @param {*} callback Callback to be executed
-	 * @memberof ResizeHandler
 	 */
-	static register(ref, callback) {
-		if (ref.isUI5Element) {
-			ref = ref.getDomRef();
+	static register(element, callback) {
+		if (element.isUI5Element) {
+			element = element.getDomRef();
 		}
 
-		ResizeHandler.attachListener(ref, callback);
+		if (element instanceof HTMLElement) {
+			observe(element, callback);
+		} else {
+			console.warn("Cannot register ResizeHandler for element", element); // eslint-disable-line
+		}
 	}
 
-
 	/**
 	 * @static
 	 * @public
-	 * @param {*} ref Reference to UI5 Web Component or DOM Element to be unobserved
-	 * @memberof ResizeHandler
+	 * @param {*} element UI5 Web Component or DOM Element to be unobserved
+	 * @param {*} callback Callback to be removed
 	 */
-	static deregister(ref, callback) {
-		if (ref.isUI5Element) {
-			ref = ref.getDomRef();
+	static deregister(element, callback) {
+		if (element.isUI5Element) {
+			element = element.getDomRef();
 		}
 
-		ResizeHandler.detachListener(ref, callback);
+		if (element instanceof HTMLElement) {
+			unobserve(element, callback);
+		} else {
+			console.warn("Cannot deregister ResizeHandler for element", element); // eslint-disable-line
+		}
 	}
 }
 
-ResizeHandler.initialize();
-
 export default ResizeHandler;
+export { setResizeHandlerObserveCallback, setResizeHandlerUnobserveCallback };
```

Comment:
Anthor suggestion - export a function to set a ResizeObserver object, and we expect that it implements the `observe` and `unobserve` methods.